File: RandoPhoto/Views/LoginActivity.cs
using System;
using Android.App;
using Android.Os;
using Android.Widget;
using Dot42;
using Dot42.Manifest;
using RandoPhoto.Presenters;
using Android.View;

namespace RandoPhoto.Views
{
    [Activity(Label = "LoginActivity", VisibleInLauncher = false)]
    public class LoginActivity : Activity, ILoginView
    {
        private EditText m_etLogin;
        private EditText m_etPassword;
        private Button m_btnLogin;
        private Button m_btnRegister;

        public LoginActivity()
        {
            m_etLogin = null;
            m_etPassword = null;
            m_btnLogin = null;
            m_btnRegister = null;

            Program.Container.ResolveToObject<ILoginView, LoginActivity>(this);
            Program.Container.Resolve(typeof(LoginPresenter));
        }

        protected override void OnCreate(Bundle savedInstance)
        {
            base.OnCreate(savedInstance);
            IViewManager viewManager = Program.Container.Resolve(typeof(IViewManager)) as IViewManager;
            viewManager.CurrentView = this;

            if (this.OnCreateViewEvent != null) this.OnCreateViewEvent(this, new EventArgs());
        }

        public void SetContent()
        {
            SetContentView(R.Layouts.login_view);
            
            m_etLogin = FindViewById<EditText>(R.Ids.etNickname);
            m_etPassword = FindViewById<EditText>(R.Ids.etPassword);
            
            m_btnLogin = FindViewById<Button>(R.Ids.btnLogin);
            m_btnLogin.Click += (sender, args) =>
            {
                LoginEventArgs loginArgs = new LoginEventArgs();
                loginArgs.UserName = m_etLogin.Text.ToString().Trim();
                loginArgs.UserPassword = m_etPassword.Text.ToString().Trim();

                if (loginArgs.UserName.IsEmpty() || loginArgs.UserPassword.IsEmpty())
                {
                    // Show error tooltip
                }
                else
                {
                    if (this.OnUserLoginClick != null) this.OnUserLoginClick(this, loginArgs);
                }
            };

            m_btnRegister = FindViewById<Button>(R.Ids.btnRegister);
            m_btnRegister.Click += (sender, args) =>
            {
                if (this.OnUserRegisterClick != null) this.OnUserRegisterClick(this, EventArgs.Empty);
            };
        }

        public void ShowLoginError(LOGINERROR loginError)
        {
            int login_text_id = -1;
            switch (loginError)
            {
                case LOGINERROR.BADPASSWORD:
                    login_text_id = R.Strings.login_error_badpassword_text;
                    break;
                case LOGINERROR.NOTEXIST:
                    login_text_id = R.Strings.login_error_notexist_text;
                    break;
                default:
                    login_text_id = R.Strings.login_error_underfined_text;
                    break;
            }

            Toast toast = Toast.MakeText(this, login_text_id, Toast.LENGTH_SHORT);
            toast.SetGravity(Gravity.CENTER, 0, 0);
            toast.Show();
        }

        public event EventHandler<EventArgs> OnCreateViewEvent;
        public event EventHandler<LoginEventArgs> OnUserLoginClick;
        public event EventHandler<EventArgs> OnUserRegisterClick;
    }
}

File: RandoApp/app/src/main/java/com/smartsoftwareagency/randophoto/LoginActivity.java
package com.smartsoftwareagency.randophoto;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.rando.library.usermanager.UserInterfaces;
import com.smartsoftwareagency.randophoto.common.ObjectFactory;

/**
 * Created by SERGant on 08.12.2014.
 */
public class LoginActivity extends Activity implements View.OnClickListener, UserInterfaces.IUserLoginCallback {
    private EditText m_etLogin;
    private EditText m_etPassword;

    public LoginActivity() {
        super();

        m_etLogin = null;
        m_etPassword = null;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        m_etLogin = (EditText) findViewById(R.id.etNickname);
        m_etPassword = (EditText) findViewById(R.id.etPassword);

        Button btnLogin = (Button) findViewById(R.id.btnLogin);
        btnLogin.setOnClickListener(this);

        Button btnRegister = (Button) findViewById(R.id.btnRegister);
        btnRegister.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch(v.getId()) {
            case R.id.btnLogin:
                loginUser();
                break;
            case R.id.btnRegister:
                registerUser();
                break;
        }
    }

    private void loginUser() {
        String login = m_etLogin.getText().toString().trim();
        String password = m_etPassword.getText().toString().trim();

        if(login.isEmpty() || password.isEmpty()) {
            showErrorMessage(R.string.login_error_nodata_text);
        }
        else {
            UserInterfaces.IUserManager userManager = ObjectFactory.getUserManager();
            userManager.LogInUser(login, password, this);
        }
    }

    private void registerUser() {
        Intent intent = new Intent(this, RegisterActivity.class);
        startActivity(intent);
    }

    private void showErrorMessage(int textID) {
        Toast toast = Toast.makeText(this, textID, Toast.LENGTH_SHORT);
        toast.setGravity(Gravity.CENTER, 0, 0);
        toast.show();
    }

    @Override
    public void OnUserLogin(UserInterfaces.IUserLoginResult iUserLoginResult) {
        UserInterfaces.LOGINRESULT loginResult = iUserLoginResult.GetUserLoginResult();

        if(loginResult == UserInterfaces.LOGINRESULT.SUCCESS) {
            Intent intent = new Intent(this, MainActivity.class);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
            startActivity(intent);
        }
        else {
            switch (loginResult) {
                case BADPASSWORD:
                    showErrorMessage(R.string.login_error_badpassword_text);
                    break;
                case NOTEXIST:
                    showErrorMessage(R.string.login_error_notexist_text);
                    break;
                case UNDEFINED:
                    showErrorMessage(R.string.login_error_underfined_text);
                    break;
            }
        }
    }
}
